dbt_rpc/rpc/builtins.py

```diff
@@ -251,8 +248,17 @@ def handle_request(self) -> PollResult:
             )
         else:
             exc = dbt.exceptions.InternalException(
-                f'Got unknown value state={state} for task {task_id}'
-            )
-            raise RPCException.from_error(
-                dbt_error(exc, logs=_dict_logs(task_logs))
+                f"Got unknown value state={state} for task {task_id}"
             )
+            raise RPCException.from_error(dbt_error(exc, logs=_dict_logs(task_logs)))
+
+
+class SigT(RemoteBuiltinMethod[SigTParameters, None]):
+    METHOD_NAME = "reload"
+
+    def set_args(self, params: SigTParameters):
+        super().set_args(params)
+
+    def handle_request(self) -> None:
+        os.kill(os.getpid(), signal.SIGHUP)
+        return os.getpid()
```

Comment:
I chose this because it ensures the most compatibility with existing behavior.

Thanks!! Renamed `SigT` to `Reload` in both class and params.

Will the reload happen synchronously:
No, I am not exactly sure whether it is a subprocess or not but this is matches exact behavior of current Sighup signal to rpc

Reply:
Awesome thank you so much!